components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java

```diff
@@ -1118,6 +1120,93 @@ public OAuthRevocationResponseDTO revokeAuthzForAppsByResourceOwner(
         return new OAuthRevocationResponseDTO();
     }
 
+    /**
+     * Revoke issued tokens for the application.
+     *
+     * @param application {@link OAuthAppRevocationRequestDTO}
+     * @return revokeRespDTO {@link OAuthAppRevocationRequestDTO}
+     * @throws IdentityOAuthAdminException Error while revoking the issued tokens.
+     */
+    public OAuthRevocationResponseDTO revokeIssuedTokensByApplication(OAuthAppRevocationRequestDTO application)
+            throws IdentityOAuthAdminException {
+
+        triggerPreApplicationTokenRevokeListeners(application);
+
+        String consumerKey = application.getConsumerKey();
+        String applicationName = application.getApplicationName();
+        if (consumerKey != null && applicationName != null) {
+            List<AccessTokenDO> activeDetailedTokens;
+            try {
+                //Get active access tokens by application id
+                activeDetailedTokens = new ArrayList<>(OAuthTokenPersistenceFactory
+                        .getInstance().getAccessTokenDAO().getActiveAcessTokenDataByConsumerKey(consumerKey));
+            } catch (IdentityOAuth2Exception e) {
+                String errorMsg = "Error occurred while retrieving access tokens issued for Client ID : " + consumerKey;
+                throw handleError(errorMsg, e);
+            }
+            if (activeDetailedTokens.size() > 0) {
+                String[] accessTokens = new String[activeDetailedTokens.size()];
+                int countToken = 0;
+                AuthenticatedUser authzUser;
+                for (AccessTokenDO detailToken : activeDetailedTokens) {
+                    String token = detailToken.getAccessToken();
+                    accessTokens[countToken++] = token;
+
+                    OAuthCacheKey cacheKeyToken = new OAuthCacheKey(token);
+                    String scope = buildScopeString(detailToken.getScope());
+                    String accessToken = detailToken.getAccessToken();
+
+                    //Clear cache with AccessTokenDO
+                    authzUser = detailToken.getAuthzUser();
+                    TokenBinding tokenBinding = detailToken.getTokenBinding();
+                    String tokenBindingReference = (tokenBinding != null &&
+                            StringUtils.isNotBlank(tokenBinding.getBindingReference())) ?
+                            tokenBinding.getBindingReference() : NONE;
+
+                    OAuthCache.getInstance().clearCacheEntry(cacheKeyToken);
+                    OAuthUtil.clearOAuthCache(detailToken.getConsumerKey(), authzUser, scope, tokenBindingReference);
```

Comment:
instead of using `detailToken.getConsumerKey()` can't we directly pass the consumer key obtained from https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1627/files#diff-6c1ece811bfc0f73bdb0d1718746f1f253119a01a268840ad862dcd3e9d077b5R1135 ?

Reply:
Addressed in [40fde89](https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1627/commits/40fde89cb099493c8f2a302b34b1141d9aea3e8d)